File: 169-Majority Element.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        res = float('inf')
        count = 0
        for i in range(len(nums)):
            if res == nums[i]:
                count += 1
            elif count == 0:
                res = nums[i]
                count = 1
            else:
                count-= 1
                
        return res

File: JavaScript/169-Majority Element.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var majorityElement = function(nums) {
    let curr = nums[0]
    let count = 0;
    for (let num of nums){
        if (num === curr) count++;
        else {
            count--;
            if (count < 0){
                curr = num;
                count = 1;
            }
        }
    }
    return curr;
};